### DISCUSSION TITLE
[A/L] (2021/22) Relazioni - Esercizio 15

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#287] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152319595-e9f20230-e415-4fdd-a1ba-c749e5b4a1bf.png)

**Comment by LuginiAndrea** [comment]
Facile notare come $R$ non sia un ordine totale in quanto non vale $9R30, 30R9, 27R30, 30R27$.
Definiamo $R^* = aRb \Longleftrightarrow aRb \vee a \leq b$
Possiamo vedere che vale la riflessività, in quanto valeva per $R$ ed inoltre ogni $a \leq a \forall a \in A$
Vale inolte l'antisimmetria, in quanto valeva in $R$ e $\forall a, b \in A, a \leq b, b \leq a \Longrightarrow b = a$
Vale la transitività: se un numero è divisibile senza resto per un altro, vuol dire che dato $D_a$ l'insieme dei fattori primi di $a$
$aRb \Longrightarrow D_a \subseteq D_b, bRc \Longrightarrow D_b \subseteq D_c$, quindi $D_a \subseteq D_b \subseteq D_c \Longleftrightarrow D_a \subseteq D_c$. Inoltre, se $a \leq b \leq c \Longrightarrow a \leq c$

Dimostriamolo con le coppie: $9 \leq 30 \Longrightarrow 9 R^* 30$, $27 \leq 30 \Longrightarrow 27 R^* 30$
Le coppie che valevano per $R$ valgono anche per $R^*$

**Comment by CiottoloMaggico** [comment]
R è una relazione di ordine parziale, in quanto:
1) Soddisfa la proprietà riflessiva: per ogni a in A esiste aRa, ogni numero può essere diviso per se stesso.
2) Soddisfa l'anti-simmetricità: per ogni a,b in A se esiste aRb e bRa allora a = b, poichè se aRb vuol dire che a >= b quindi non può valere bRa a meno che a = b
3) Soddisfa la transitività: se esiste aRb e bRc => aRc, in quanto se aRb e bRc => c è uno dei divisori di a
4) Non soddisfa la totalità: per via dell'esistenza di coppie di incomparabili

Quindi possiamo procedere estendendo la relazione ad un'ordine totale come segue:


![immagine](https://user-images.githubusercontent.com/79638284/209202997-eadc5252-7cca-4dff-b35d-cd3138f8529b.png)
![immagine](https://user-images.githubusercontent.com/79638284/209203026-17193a6d-557c-473f-9988-f8cdad3cd3d2.png)
![immagine](https://user-images.githubusercontent.com/79638284/209203079-761c1a54-194d-480a-a465-fd2353937ecb.png)
![immagine](https://user-images.githubusercontent.com/79638284/209203099-6eac1e24-ca7e-4259-9f2f-2f0bd5649b0b.png)
